community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnPage.java

```diff
@@ -326,7 +326,7 @@ public int swapIn( StoreChannel channel, long offset, int length ) throws IOExce
             while ( read != -1 && (readTotal += read) < length );
 
             // Zero-fill the rest.
-            UnsafeUtil.setMemory( pointer + readTotal, getCachePageSize() - readTotal, (byte) 0 );
+            UnsafeUtil.setMemory( pointer + readTotal, length - readTotal, (byte) 0 );
```

Comment:
From what I can understand `getCachePageSize()` is the actual size, including the "accidental" bytes in the end due to misaligned record sizes, and `length` is the page size without the accidental bytes in the end. Correct?